src/utils/pgwire/src/pg_protocol.rs

```diff
@@ -153,15 +153,40 @@ pub fn cstr_to_str(b: &Bytes) -> Result<&str, Utf8Error> {
 
 /// Record `sql` in the current tracing span.
 fn record_sql_in_span(sql: &str) {
+    let redacted_sql = redact_sql(sql);
     tracing::Span::current().record(
         "sql",
         tracing::field::display(truncated_fmt::TruncatedFmt(
-            &sql,
+            &redacted_sql,
             *RW_QUERY_LOG_TRUNCATE_LEN,
         )),
     );
 }
 
+fn redact_sql(sql: &str) -> String {
+    match Parser::parse_sql(sql) {
+        Ok(sqls) => sqls
+            .into_iter()
+            .map(|sql| sql.to_redacted_string())
+            .join(";"),
+        Err(_) => {
```

Comment:
Actually I also find this not that necessary. 😕 We can just leave the statements that failed to parse as they are.


In other words, what if users misspell "with" as "wth"? It appears that we are unable to handle all cases.

Reply:
fixed